### ISSUE TITLE
I get negative coordinate values after rescale box

### ISSUE THREAD

**Issue by ChengKai-Wang** [eriklindernoren/PyTorch-YOLOv3#640]
My original image size is 680 * 480
then use pytorch transform rescale to 416*416 and put into YOLO network
After non_max_supression, x, y, x, y are still postive.
But after rescale_boxes , the results become negative.
How to solve this?
Thank you.

**Comment by ChengKai-Wang** [comment]
I set pad_x =0 and pad_y = 0
then it looks right.
Is this operation OK?

**Comment by Flova** [comment]
DO you use the current master? At what point of the pipeline are you applying the transform? There are multiple mandatory transforms that also to rescaling in the pipeline and the sequence of them matters.

**Comment by ChengKai-Wang** [comment]
> DO you use the current master? At what point of the pipeline are you applying the transform? There are multiple mandatory transforms that also to rescaling in the pipeline and the sequence of them matters.

At inference stage. I wrote a detection.py  as following (partial code):
detection = Darknet("config/yolov3.cfg", img_size = (480, 640)).to(device)
detection.load_state_dict(torch.load("weight.pth", map_location="cpu"))
img = cv2.imread('./test.jpg')
img = PIL.Image.fromarray(cv2.cvtColor(img, cv2.COLOR_BGR2RGB))
transform = torchvision.transforms.Compose([torchvision.transforms.Resize((416,416), interpolation=2), torchvision.transforms.ToTensor()])
img = transform(img)
results = detection(img)
results = non_max_suppression(results, 0.8, 0.3)
results = rescale_boxes(results, 416, (480, 640))

**Comment by Flova** [comment]
Ah okay. Yolo expects the images to be padded with black to a square. You are squeezing to image to 416x416 using your resize operation. The function `rescale_boxes` expects the image to be padded. I would suggest using the standard padding, which is included in the `DEFAULT_TRANSFORMS` transform compose. 

Example:
`transform = transforms.Compose([DEFAULT_TRANSFORMS, Resize(416)]))`

**Comment by ChengKai-Wang** [comment]
Thank U!
